product_variant_configurator/models/product_configurator.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, exceptions, _
+from openerp.addons import decimal_precision as dp
+
+
+class ProductConfigurator(models.AbstractModel):
+    _name = 'product.configurator'
+
+    product_tmpl_id = fields.Many2one(
+        string='Product Template',
+        comodel_name='product.template',
+        auto_join=True)
+    product_attribute_ids = fields.One2many(
+        comodel_name='product.configurator.attribute',
+        domain=lambda self: [("owner_model", "=", self._name)],
+        inverse_name='owner_id', string='Product attributes', copy=True)
```

Comment:
Using an Integer field as inverse_name of a One2many field crashes when deleting records from the One2many, because the ORM searches for an `ondelete` attribute on the field, which only is on Many2one fields.